photon-ml/src/main/scala/com/linkedin/photon/ml/io/LibSVMInputDataFormat.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.io
+
+import breeze.linalg.SparseVector
+import com.linkedin.photon.ml.data.LabeledPoint
+import com.linkedin.photon.ml.util.{IdentityIndexMapLoader, IndexMapLoader}
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+
+/**
+  * An input format supporting reading LibSVM txt, the assumed format is:
+  * [label] [feature_index]:[feature_value] [feature_index]:[feature_value] ....
+  * (delimited by space)
+  *
+  * By default, libsvm feature indices start from 1, but if zeroBased is true, we will assume it
+  * starts from 0.
+  */
+class LibSVMInputDataFormat(val featureDimension: Int,
+                            val useIntercept: Boolean = true,
+                            val zeroBased: Boolean = false,
+                            val delim: String = " ",
+                            val idxValueDelim: String = ":"
+                           ) extends InputDataFormat {
```

Comment:
Please make sure that changes follow the Scala style guide:
http://docs.scala-lang.org/style/declarations.html#classes